app/models/chat_channel_membership.rb

```diff
@@ -15,7 +15,8 @@ class ChatChannelMembership < ApplicationRecord
   validates :role, inclusion: { in: %w[member mod] }
   validate  :permission
 
-  after_commit :index_to_elasticsearch
+  after_commit :index_to_elasticsearch, on: %i[create update]
+  after_commit :remove_from_elasticsearch, on: [:destroy]
```

Comment:
as they seem they are going be repeated in all models maybe we can move them in the `Searchable` module

Reply:
I thought about putting these callbacks in there but I wanted more control in the models for when indexing happens since it might not be after every commit for every model. This may come into play for searchables (user, article, podcast episode)